Source/WPEFramework/PluginServer.cpp

```diff
@@ -724,14 +734,21 @@ namespace PluginHost {
                         WakeupProcess(timeout, parentPID, _administrator.Configuration().HibernateLocator().c_str(), _T(""), &_hibernateStorage);
                     }
                 }
+
+                if (result == Core::ERROR_NONE) {
+                    SYSLOG(Logging::Startup, ("Hibernated plugin [%s]:[%s]", ClassName().c_str(), Callsign().c_str()));
+                }
+                Lock();
 #else
+                local->Release();
                 result = Core::ERROR_NONE;
 #endif
                 if (result == Core::ERROR_NONE) {
-                    State(IShell::state::HIBERNATED);
                     SYSLOG(Logging::Notification, ("Hibernated plugin [%s]:[%s]", ClassName().c_str(), Callsign().c_str()));
```

Comment:
This probably needs a change of category as I assume that this reports that the request to hibernate (from the outside interface) has succeeded?? So I guess this should be logging:Startup ?